### ISSUE TITLE
Trying to watch video produces error for some

### ISSUE THREAD

**Issue by eddierubeiz** [sciencehistory/scihist_digicoll#2404]
Visit
https://digital.sciencehistory.org/works/wi609jf

For some people, the video is not playable, instead of they get an error displayed in video player: "Media could not be loaded, either because the server or network failed or because the format is not supported."

Those encountering this error can still play other videos. 

Eddie did encounter this error reorted in Web dev console:

> wi609jf:1 Access to XMLHttpRequest at 'https://d2wxoyqt1kxdpi.cloudfront.net/hls/9df2dfe8-3ce5-45a5-920f-dc6b1e67f01b/b45847d617307b2242e074870c3637e2/hls.m3u8' from origin 'https://digital.sciencehistory.org/' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

For Jonathan the video displays fine.

**Comment by eddierubeiz** [comment]
Reproducible on latest version of Chrome, but not on Safari.

**Comment by jrochkind** [comment]
if I try that URL, I personally DO see those CORS headers... although they are all in lower-case. Which is what cloudfront does.  And I think is supposed to be equivalent.

```
curl -I https://d2wxoyqt1kxdpi.cloudfront.net/hls/9df2dfe8-3ce5-45a5-920f-dc6b1e67f01b/b45847d617307b2242e074870c3637e2/hls.m3u8
HTTP/2 200
content-type: application/vnd.apple.mpegurl
content-length: 343
date: Thu, 05 Oct 2023 18:49:36 GMT
access-control-allow-origin: *
access-control-allow-methods: GET
access-control-max-age: 43200
last-modified: Thu, 05 Oct 2023 17:11:12 GMT
etag: "a2b6bfacd3852516f1b72a8c736f8ae8"
x-amz-server-side-encryption: AES256
x-amz-version-id: 2cO6TucgGF_wi8LocgORGqozqJqB1dB_
accept-ranges: bytes
server: AmazonS3
x-cache: Hit from cloudfront
via: 1.1 bad10a325a27114250a7d64ce1bcf6ac.cloudfront.net (CloudFront)
x-amz-cf-pop: IAD66-C2
x-amz-cf-id: DLr99A7Bo32o7EY1KWjGNjsE5ai7EJ4itK9yVhEE3jb2dJr5kiDIWQ==
age: 1450
cache-control: max-age=31536000, immutable
```

I am not necessarily suspecting the case difference is to blame.

I think it's equally likely that sometimes the headers are not being returned. 

This is set to cache forever, so once you get one that works (or doesn't), I think it'll "stick".

**Comment by eddierubeiz** [comment]
```
curl -I https://d2wxoyqt1kxdpi.cloudfront.net/hls/9df2dfe8-3ce5-45a5-920f-dc6b1e67f01b/b45847d617307b2242e074870c3637e2/hls.m3u8
HTTP/2 200
content-type: application/vnd.apple.mpegurl
content-length: 343
date: Thu, 05 Oct 2023 17:34:36 GMT
last-modified: Thu, 05 Oct 2023 17:11:12 GMT
etag: "a2b6bfacd3852516f1b72a8c736f8ae8"
x-amz-server-side-encryption: AES256
x-amz-version-id: 2cO6TucgGF_wi8LocgORGqozqJqB1dB_
accept-ranges: bytes
server: AmazonS3
x-cache: Hit from cloudfront
via: 1.1 c9bef6d423a5d23e0ca5e2af8503331c.cloudfront.net (CloudFront)
x-amz-cf-pop: EWR53-C1
x-amz-cf-id: 9NaZoe_fOuhNg6-6zO-K7FdOGwLEeKI1689Cd7muC_D44RYy8yENQA==
age: 6424
cache-control: max-age=31536000, immutable
```

**Comment by jrochkind** [comment]
If Cloudfront is delivering different headers to different people... that's odd.  We do see in the responses that we have different `x-amz-cf-pop`s, I guess they are caching different things?  

I wonder if it could be a bug in CloudFront. 

I don't know if we have CloudFront itself adding the headers, or forwarding them them from the S3 bucket. If forwarding from the S3 bucket, it could be a bug/problem in the S3 bucket if sometimes it is returning the headers but not other times.

**Comment by jrochkind** [comment]
The original S3 object is NOT returning CORS headers for me. 

```
curl -I "https://scihist-digicoll-production-derivatives-video.s3.amazonaws.com/hls/9df2dfe8-3ce5-45a5-920f-dc6b1e67f01b/b45847d617307b2242e074870c3637e2/hls.m3u8"
HTTP/1.1 200 OK
x-amz-id-2: uKORzm3Sh0X5GO3fnjUW6EXy1hEKo5ajLR9aBKN5DNBzPdZt9WJpZ79CjECGVC2bqsWrwuvVjPE=
x-amz-request-id: 5ZTVDW1Y352GW48K
Date: Thu, 05 Oct 2023 19:33:07 GMT
Last-Modified: Thu, 05 Oct 2023 17:11:12 GMT
ETag: "a2b6bfacd3852516f1b72a8c736f8ae8"
x-amz-server-side-encryption: AES256
x-amz-version-id: 2cO6TucgGF_wi8LocgORGqozqJqB1dB_
Accept-Ranges: bytes
Content-Type: application/vnd.apple.mpegurl
Server: AmazonS3
Content-Length: 343
```

I don't see any configuration in the cloudfront distribution that is adding them. I'm not sure how we were getting them before, for sure. 

Hypothesis:

1. At some point in the past the S3 bucket WAS returning CORS headers. 
2. Cloudfront passses them on
3. As a result of a change (we have edited S3 buckets in TF in the past weeks/months), they stopped adding CORS headers
4. Since S3 delivers things with far-future-cacheable headeres, some CloudFront hosts are still returning the previous cached copy with CORS headers, others are getting a fresh copy without CORS headers.

**Comment by eddierubeiz** [comment]
Note: the item was ingested this afternoon and worked for everyone here in the building for the first few minutes.

**Comment by jrochkind** [comment]
S3 seems not to be returning CORS headers. One omission from my previous test -- in order to get the CORS headers, you are expected to have to have an "Origin" header in request. 

So trying again with that:

```
$ curl -H "Origin: https://staging-digital.sciencehistory.org"   -I "https://scihist-digicoll-staging-derivatives-video.s3.amazonaws.com/hls/a26379fe-f2cf-4e9f-b178-bc73daa6cdda/6a0b6651ca70c380805880d838b02f4b/hls.m3u8"
HTTP/1.1 200 OK
x-amz-id-2: iZezIGUPcfDUMY0S3neaERdghf4cqHZuXN5UovPdUwt40vfrvniJUVHmW22g4vfmWkKUYS6v4wc=
x-amz-request-id: 271Q2C6T1PMCH26P
Date: Thu, 05 Oct 2023 20:06:51 GMT
Last-Modified: Tue, 13 Jun 2023 14:26:06 GMT
ETag: "a3ec56cbe52da0f3d8e887b8133e9685"
x-amz-server-side-encryption: AES256
x-amz-version-id: WZpYOn_tpIqcxA8G71yN20Rkk9qERKdB
Accept-Ranges: bytes
Content-Type: application/vnd.apple.mpegurl
Server: AmazonS3
Content-Length: 343
```

Still no CORS headers. 

Yet the `scihist-digicoll-staging-derivatives-video.s3.amazonaws.com` bucket has a CORS policy set:

```
[
    {
        "AllowedHeaders": [
            "*"
        ],
        "AllowedMethods": [
            "GET"
        ],
        "AllowedOrigins": [
            "*"
        ],
        "ExposeHeaders": [],
        "MaxAgeSeconds": 43200
    }
]
```

Changing AllowedOrigins to specifically be `https://staging-digital.sciencehistory.org` did not improve things either. 

Do we have something wrong in our CORS config? But this used to work reliably... is S3 malfunctioning?

**Comment by jrochkind** [comment]
In the bucket, If I add `HEAD` to the CORS config allowed methods too:

```
        "AllowedMethods": [
            "GET",
            "HEAD"
        ],
```

Then I start getting the headers in the response?

```
curl -H "Origin: https://staging-digital.sciencehistory.org"   -I "https://scihist-digicoll-staging-derivatives-video.s3.amazonaws.com/hls/a26379fe-f2cf-4e9f-b178-bc73daa6cdda/6a0b6651ca70c380805880d838b02f4b/hls.m3u8"
HTTP/1.1 200 OK
x-amz-id-2: ZvTJLbGrHl5PL+QZfZ99fNq1NOhTtEUQJFUsuE8moG8yrwZFMRxQCYkVExpjBJjNC6mFhFUwrFYs4rocKxTwEA==
x-amz-request-id: FC633DNHJP3S4XF4
Date: Thu, 05 Oct 2023 20:20:14 GMT
Access-Control-Allow-Origin: *
Access-Control-Allow-Methods: GET, HEAD
Access-Control-Max-Age: 43200
Vary: Origin, Access-Control-Request-Headers, Access-Control-Request-Method
Last-Modified: Tue, 13 Jun 2023 14:26:06 GMT
ETag: "a3ec56cbe52da0f3d8e887b8133e9685"
x-amz-server-side-encryption: AES256
x-amz-version-id: WZpYOn_tpIqcxA8G71yN20Rkk9qERKdB
Accept-Ranges: bytes
Content-Type: application/vnd.apple.mpegurl
Server: AmazonS3
Content-Length: 343
```

Not sure why this is necessary... or why it would have just become necessary! 

Doing some googling to see what methods we should want to add there.

**Comment by jrochkind** [comment]
Maybe add `OPTIONS` too while we're at it. I think `GET`, `HEAD` and `OPTIONS` might be the way to go... very confusing though.

**Comment by eddierubeiz** [comment]
You were right -- this fixed it!
I just went ahead and self-merged the change in Terraform. Made the change in both staging and prod, invalidated both caches.
Thanks again!

**Comment by eddierubeiz** [comment]
This is recurring with [this video](https://digital.sciencehistory.org/works/9nhk6as).

**Comment by jrochkind** [comment]
@eddierubeiz we need more details than "this or something like it" -- who reported this, and what were they experiencing, what happened that was different than they wanted?   Is it the same error, "Media could not be loaded, either because the server or network failed or because the format is not supported"?  Is it reproducible for them? (It is not for me) Is it someone at SHI where we can go look at their computer as they do it, and look at Chrome dev tools and see what errors are?

**Comment by jrochkind** [comment]
Re-closing in favor of a separate ticket starting over, to keep things readable, which we link together. New ticket at #2523
